src/StatsdClient/StatsdBuilder.cs

```diff
@@ -0,0 +1,123 @@
+using System;
+using System.Net;
+using Mono.Unix;
+using StatsdClient.Bufferize;
+
+namespace StatsdClient
+{
+    /// <summary>
+    /// StatsdBuilder builds an instance of `Statsd` from StatsdConfig.
+    /// </summary>
+    class StatsdBuilder : IDisposable
+    {
+        public static readonly string UnixDomainSocketPrefix = "unix://";
+
+        readonly IStatsBufferizeFactory _factory;
+
+        StatsSender _statsSender;
+        StatsBufferize _statsBufferize;
+
+
+        public StatsdBuilder(IStatsBufferizeFactory factory)
+        {
+            _factory = factory;
+        }
+
+        public Statsd BuildStats(StatsdConfig config)
+        {
+            var statsdServerName = !string.IsNullOrEmpty(config.StatsdServerName)
```

Comment:
I kept the existing behavior to avoid a breaking change:
https://github.com/DataDog/dogstatsd-csharp-client/blob/master/src/StatsdClient/StatsdUDP.cs#L31-L34.

Do you think I should update it now?

Reply:
No, let's keep it as it is. We should discuss with the team, because I think different clients behave differently. I think python, java, go, C# should probably all behave the same way.